app/scripts/views/mixins/settings-mixin.js

```diff
@@ -0,0 +1,37 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// helper functions for views with a profile image. Meant to be mixed into views.
+
+'use strict';
+
+define([
+  'lib/session'
+], function (Session) {
+
+  return {
+    // user must be authenticated and verified to see Settings pages
+    mustVerify: true,
+
+    initialize: function () {
+      var self = this;
+      var uid = self.searchParam('uid');
+
+      // A uid param is set by RPs linking directly to the settings
+      // page for a particular account.
+      // We set the current account to the one with `uid` if
+      // it exists in our list of cached accounts. If it doesn't,
+      // clear the current account.
```

Comment:
The current UX for the settings page, which this borrows, does not force an email address.